sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -83,14 +87,13 @@ trait ExpressionWithRandomSeed {
   """,
   since = "1.5.0")
 // scalastyle:on line.size.limit
-case class Rand(child: Expression, hideSeed: Boolean = false)
-  extends RDG with ExpressionWithRandomSeed {
+case class Rand(child: Expression, hideSeed: Boolean = false) extends RDG {
 
-  def this() = this(Literal(Utils.random.nextLong(), LongType), true)
+  def this() = this(UnresolvedSeed, true)
 
   def this(child: Expression) = this(child, false)
 
-  override def withNewSeed(seed: Long): Rand = Rand(Literal(seed, LongType))
+  override def withNewSeed(seed: Long): Rand = Rand(Literal(seed, LongType), hideSeed)
```

Comment:
Ur, is this related to this PR, @ulysses-you ? This looks like an orthogonal bug fix a little to me. I'm wondering if we we have a test coverage for this.

Reply:
@dongjoon-hyun thanks for the careful check. Yes, it could be thought as bug fix. See the [jenkins test result](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/133209/testReport/org.apache.spark.sql/SQLQuerySuite/SPARK_31594__Do_not_display_the_seed_of_rand_randn_with_no_argument_in_output_schema/) before this commit [a3c9f0b](https://github.com/apache/spark/pull/30864/commits/a3c9f0b1410f0b3b367c70dc534b1c0d0b08e137).

And I think for early branch, we just need to backport this place and leave this PR for master. But also, I'm OK to send a PR to fix this first, then go back to this PR.

Reply:
This is an orthogonal bug fix, but it only affects the `sql` method. I don't have a strong opinion about backporting the fix or not.

Reply:
Got it. Thanks, @cloud-fan .